packages/generator/src/generator.ts

```diff
@@ -150,6 +184,10 @@ export abstract class Generator<
 
   abstract getTargetDirectory(): string
 
+  getPromptRunner(): PromptRunner | undefined {
+    return undefined
```

Comment:
suggestion: return an empty prompt runner here `new PromptRunner([])` to avoid needing to return undefined. It makes the interface a bit cleaner and removes the need to have optional chaining everywhere.